subxt/src/custom_values/custom_value_address.rs

```diff
@@ -38,13 +38,25 @@ pub struct Yes;
 
 /// A static address to a custom value.
 #[derive(Derivative)]
-#[derivative(Clone(bound = ""), Debug(bound = ""))]
+#[derivative(
+    Clone(bound = ""),
+    Debug(bound = ""),
+    Eq(bound = ""),
+    Ord(bound = ""),
+    PartialEq(bound = "")
+)]
 pub struct StaticAddress<ReturnTy, IsDecodable> {
     name: &'static str,
     hash: Option<[u8; 32]>,
     phantom: PhantomData<(ReturnTy, IsDecodable)>,
 }
 
+impl<ReturnTy, IsDecodable> PartialOrd for StaticAddress<ReturnTy, IsDecodable> {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
```

Comment:
Interesting; it sortof looks like derivative would do this (https://docs.rs/derivative/latest/src/derivative/lib.rs.html#61) already, but I guess you ran into an issue using it?

Reply:
For this avoid the [clippy warning](https://github.com/paritytech/subxt/actions/runs/7871592798/job/21475224872), the derivative crate would need to take into account the suggestion from https://rust-lang.github.io/rust-clippy/master/index.html#/non_canonical_partial_ord_impl.
Which is to delegate the `PartialOrd::partial_cmp` implementation, to the `Ord::cmp`; I don't think that it does that by the looks of those functions.

Probably the clippy rule has been added for implementations without bounds, not sure why it wasn't triggered for our other derives.

Reply:
Aah ok, basically this issue: https://github.com/mcarton/rust-derivative/issues/115

Probably needs derivative to emit that clippy allow in the relevant place!